src/image/ExposureFitsReader.cc

```diff
@@ -48,8 +52,155 @@ bool _contains(std::array<T, N> const& array, T const& value) {
     return false;
 }
 
+// Map from compatibility "afw name" to correct filter label
+std::map<std::string, FilterLabel> const _AFW_NAMES = {
+        std::make_pair("r2", FilterLabel::fromBandPhysical("r", "HSC-R2")),
+        std::make_pair("i2", FilterLabel::fromBandPhysical("i", "HSC-I2")),
+        std::make_pair("SOLID", FilterLabel::fromPhysical("solid plate 0.0 0.0")),
+};
+
+/**
+ * Determine heuristically whether a filter name represents a band or a physical filter.
+ *
+ * @param name The name to test.
+ */
+bool _isBand(std::string const& name) {
+    static std::set<std::string> const BANDS = {"u", "g", "r", "i", "z", "y", "SH", "PH", "VR", "white"};
+    // Standard band
+    if (BANDS.count(name) > 0) {
+        return true;
+    }
+    // Looks like a narrow-band band
+    if (std::regex_match(name, std::regex("N\\d+"))) {
```

Comment:
Subaru also has `I945`, which I'm guessing might be "intermediate-band" (which I'd normally call "medium band"). Not sure if that's worth a check here too?

Reply:
Good catch, though I'll have to be careful to exclude "I2".